pkg/shared/families/malware/yara/util/yara_util.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2023 Cisco Systems, Inc. and its affiliates.
+// All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/openclarity/vmclarity/pkg/shared/families/malware/common"
+)
+
+const (
+	// A yara detected malware line should be in the following format: <yara rule name> [<coma,separated,meta data>] <path>.
+	scanError                   = "error scanning"
+	yaraRuleNamePosition        = 0
+	yaraRuleMetadataPosition    = 1
+	detectedMalwarePathPosition = 2
+	malwareNameString           = "name" // The key defined in the metadata of yara rule
+	malwareTypeString           = "type" // The key defined in the metadata of yara rule
+	yaraOutputFiledNum          = 3      // Number of yara output fields. <yara rule name> [<coma,separated,meta data>] <path>
+)
+
+func ParseYaraScanOutput(yaraOutput string, logger *logrus.Entry) []common.DetectedMalware {
+	logger.Debugf("Parsing yara output: %s", yaraOutput)
+	lines := strings.Split(yaraOutput, "\n")
+
+	malwareInfoList := make([]common.DetectedMalware, 0)
+	for i := range lines {
+		logger.Debugf("line %d: %s", i, lines[i])
+		if strings.Contains(lines[i], scanError) {
+			logger.Errorf("Yara scan failed: %s", lines[i])
+			continue
+		}
+
+		words := separateFields(lines[i])
+		// If the line has less than `detectedMalwareLineWordCount` words in it, it can't be a malware detection, so it's skipped
+		if words == nil {
+			logger.Debugf("omitting invalid yara line: %s", words)
+			continue
+		}
+
+		detectedMalware := extractDetectedMalware(words)
+		malwareInfoList = append(malwareInfoList, *detectedMalware)
+	}
+
+	return malwareInfoList
+}
+
+func separateFields(line string) []string {
+	words := make([]string, yaraOutputFiledNum)
+	fields := strings.Split(line, "[")
+	if len(fields) != 2 { // nolint:gomnd
+		return nil
+	}
+	words[yaraRuleNamePosition] = strings.TrimSpace(fields[0])
+	fields = strings.Split(fields[1], "]")
+	if len(fields) != 2 { // nolint:gomnd
+		return nil
+	}
+	words[yaraRuleMetadataPosition] = fields[0]
+	words[detectedMalwarePathPosition] = strings.TrimSpace(fields[1])
+
+	return words
+}
```

Comment:
It might be clearer to use strings.Cut for example:
```
ruleName, remainder, found := strings.Cut(line, "[")
if !found {
    return nil
}

metadata, path, found := strings.Cut(remainder, "]")
if !found {
   return nil
}

return []string{strings.TrimSpace(ruleName), strings.TrimSpace(metadata), strings.TrimSpace(path)}
```

Perhaps we also want to return a struct here instead of a list:

```
type ScanResultLine struct {
    RuleName string
    Metadata string
    Path string
}
```

You could then parse the metadata into KV as part of this function and return:

```
type ScanResultLine struct {
    RuleName string
    Metadata map[string]string
    Path string
}
```